### PR TITLE
Turn param types from docblock to typehints

### LINKED ISSUE #376: Inconsistent annotations of constructors
### What is happening
Right now in our application there is inconsistency in our `__construct` methods. Sometimes we do not write annotations and sometimes we do.

### Expected result
There is decision on if we want to write or do not want to write annotations to constructors and annotations are changed on this decision

### PR DESCRIPTION
**Pull request by TomasVotruba**
| Q             | A
| ------------- | ---
|New feature| No
|BC breaks| No
|Fixes issues| Closes #376
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

As response to an email by @Miroslav-Stopka  :)

Inspiration: https://github.com/kukulich/php-type-hints-convertor

**This is just prototype to discuss over. It needs to run tests and fix the some missing typehints due to inconsistent parent classes/interfaces.**

### How to Replicate?

```yaml
# easy-coding-standard.yml
services:
    PhpCsFixer\Fixer\FunctionNotation\PhpdocToReturnTypeFixer:

    SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff:
        usefulAnnotations:
            - '@deprecated'
            - '@see'
            - '@dataProvider'
            - '@group'

    Symplify\CodingStandard\Fixer\Commenting\RemoveUselessDocBlockFixer:


parameters:
    skip:
        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingTraversableParameterTypeHintSpecification: ~

        # remove, run then add
        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingParameterTypeHint: ~

        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingTraversableReturnTypeHintSpecification: ~
        
        # remove, run then add
        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingReturnTypeHint: ~

        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingTraversablePropertyTypeHintSpecification: ~

        SlevomatCodingStandard\Sniffs\TypeHints\TypeHintDeclarationSniff.MissingPropertyTypeHint: ~
```
	
**dry-run**

```bash
vendor/bin/ecs check packages project-base 
```

**fix**

```bash
vendor/bin/ecs check packages project-base --fix
```

### CODE REVIEW

**Comment by TomasVotruba** [comment]
WIP...

File: packages/coding-standards/src/Sniffs/ForbiddenSuperGlobalSniff.php [hunk 1]
```diff
@@ -27,11 +27,8 @@ public function register(): array
     {
         return [T_VARIABLE];
     }
-
-    /**
-     * @param int $position
-     */
-    public function process(File $file, $position): void
+    
+    public function process(File $file, int $position): void
```
Comment (TomasVotruba) [inline packages/coding-standards/src/Sniffs/ForbiddenSuperGlobalSniff.php]: Should be fixed

File: packages/coding-standards/src/Sniffs/ObjectIsCreatedByFactorySniff.php [hunk 2]
```diff
@@ -28,12 +28,8 @@ public function register(): array
...
         return [T_NEW];
     }
-
-    /**
-     * @param \PHP_CodeSniffer\Files\File $file
-     * @param int $position
-     */
-    public function process(File $file, $position): void
+    
+    public function process(File $file, int $position): void
```
Comment (TomasVotruba) [inline packages/coding-standards/src/Sniffs/ObjectIsCreatedByFactorySniff.php]: Should be fixed

File: packages/coding-standards/src/Sniffs/ValidVariableNameSniff.php [hunk 3]
```diff
@@ -19,26 +19,17 @@
  */
 final class ValidVariableNameSniff extends AbstractVariableSniff
 {
-    /**
-     * @param int $position
-     */
-    protected function processVariable(File $file, $position): void
+    protected function processVariable(File $file, int $position): void
```
Comment (TomasVotruba) [inline packages/coding-standards/src/Sniffs/ValidVariableNameSniff.php]: Should be fixed

**Comment by Miroslav-Stopka** [comment]
Hi @TomasVotruba,

thank you for this pull request :). We have decided to solve this problem somewhat systematically. We need to explore it more deeply and decide how we want to work with annotations and, in fact, with the type hints.

This pull request will be used as inspiration but for now, I'm closing it.

**Comment by TomasVotruba** [comment]
Sure. I look forward to upcoming code cleanups :+1:

**Comment by Miroslav-Stopka** [comment]
Hi @TomasVotruba,
we discussed it with several developers, and we decided to annotate everything. This is the most standard way, and at the same time makes the annotation clear and consistent. Problem will be solved in #417.

**Comment by TomasVotruba** [comment]
Hey Miro, no troubles. I know you've decided what's best for the team.

Let me know if you need help with the fixer for this.

**Pull request closed without merging**